lib/datastore/request.js

```diff
@@ -148,6 +148,15 @@ DatastoreRequest.prototype.get = function(keys, callback) {
 };
 
 /**
+ * Maps to {module:datastore/dataset#save}, forcing the method to be `insert`.
+ */
+DatastoreRequest.prototype.insert = function(entities, callback) {
+  var isArray = util.is(entities, 'array');
+  entities = util.arrayize(entities).map(util.propAssign('method', 'insert'));
+  this.save(isArray ? entities : entities[0], callback);
```

Comment:
We shouldn't need any of this logic, it gets put into an array in `save` anyway. Snippet:

``` js
DatastoreRequest.prototype.save = function(entities, callback) {
  var isMultipleRequest = Array.isArray(entities);
  entities = isMultipleRequest ? entities : [entities];
  // ...
```
Actually, while we are here, mind using `util.arrayize()` in `save()` instead of those two lines?